test/util/server/server.go

```diff
@@ -358,19 +361,41 @@ func StartConfiguredMasterWithOptions(masterConfig *configapi.MasterConfig, test
 		return "", err
 	}
 
-	for {
-		// confirm that we can actually query from the api server
-		if client, err := util.GetClusterAdminClient(adminKubeConfigFile); err == nil {
-			if _, err := client.ClusterPolicies().List(metav1.ListOptions{}); err == nil {
-				break
-			}
+	err = wait.Poll(100*time.Millisecond, 1*time.Minute, func() (bool, error) {
+		var healthy bool
+		healthy, err = IsServerHealthy(*masterURL)
+		if err != nil {
+			return false, err
 		}
-		time.Sleep(100 * time.Millisecond)
+		return healthy, nil
+	})
+	if err == wait.ErrWaitTimeout {
+		return "", fmt.Errorf("server did not become healthy")
+	}
+	if err != nil {
+		return "", err
 	}
 
 	return adminKubeConfigFile, nil
 }
 
+func IsServerHealthy(url url.URL) (bool, error) {
+	transport := knet.SetTransportDefaults(&http.Transport{
+		TLSClientConfig: &tls.Config{
+			InsecureSkipVerify: true,
+		},
+	})
+
+	url.Path = "/healthz"
```

Comment:
the RBAC post-start hook will actually not get enabled once we pick up https://github.com/kubernetes/kubernetes/pull/43920